tests/trial_tests/test_trial.py

```diff
@@ -450,25 +472,32 @@ def test_suggest_int_invalid_step() -> None:
 @pytest.mark.parametrize("storage_mode", STORAGE_MODES)
 def test_suggest_int_log(storage_mode: str) -> None:
 
-    mock = Mock()
-    mock.side_effect = [1, 2]
-    sampler = samplers.RandomSampler()
+    low = 1
+    high = 3
+    x = 1
+    y = 2
 
-    with patch.object(sampler, "sample_independent", mock) as mock_object, StorageSupplier(
-        storage_mode
-    ) as storage:
+    relative_search_space = {
+        "x": IntDistribution(low=low, high=high, log=True),
+        "y": IntDistribution(low=low, high=high, log=True),
+    }
+    relative_params = {"x": x, "y": y}
+    sampler = DeterministicRelativeSampler(relative_search_space, relative_params)  # type: ignore
+
+    with StorageSupplier(storage_mode) as storage:
         study = create_study(storage=storage, sampler=sampler)
         trial = Trial(study, study._storage.create_new_trial(study._study_id))
-        distribution = IntDistribution(low=1, high=3, log=True)
 
-        assert trial._suggest("x", distribution) == 1  # Test suggesting a param.
-        assert trial._suggest("x", distribution) == 1  # Test suggesting the same param.
-        assert trial._suggest("y", distribution) == 2  # Test suggesting a different param.
-        assert trial.params == {"x": 1, "y": 2}
-        assert mock_object.call_count == 2
+        assert trial.suggest_int("x", low, high, log=True) == x  # Test suggesting a param.
+        assert trial.suggest_int("x", low, high, log=True) == x  # Test suggesting the same param.
+        assert (
+            trial.suggest_int("y", low, high, log=True) == y
+        )  # Test suggesting a different param.
+        assert trial.params == {"x": x, "y": y}
 
+    sampler1 = samplers.RandomSampler()
```

Comment:
How about splitting this test case into two? It looks that this test verifies too much functionality in a single test method.

Reply:
ok. Divided it.